langwatch/src/components/AddTriggerDrawer.tsx

```diff
@@ -59,24 +78,117 @@ export function TriggerDrawer() {
 
   const { filterParams } = useFilterParams();
 
+  const [localStorageDatasetId, setLocalStorageDatasetId] =
+    useLocalStorage<string>("selectedDatasetId", "");
+
   const {
     register,
     handleSubmit,
     watch,
     formState: { errors },
     reset,
+    setValue,
   } = useForm({
     defaultValues: {
       name: "",
       action: TriggerAction.SEND_EMAIL,
       email: "",
       members: [],
       slackWebhook: "",
+      datasetId: localStorageDatasetId,
     },
   });
 
+  const datasetId = watch("datasetId");
+
+  const selectedDataset = datasets.data?.find(
+    (dataset) => dataset.id === datasetId
+  );
+
+  const tracesWithSpans = api.traces.getSampleTracesDataset.useQuery(
+    {
+      ...filterParams,
+      projectId: project?.id ?? "",
+    },
+    {
+      enabled: !!project,
+      refetchOnWindowFocus: false,
+    }
+  );
+
+  const columnDefs = useMemo(() => {
```

Comment:
is this something that could go inside the DatasetMappingPreview? This way we wouldn't need to duplicate the definitions. We already need to send the columnTypes down anyway